shared/src/panel/views/HierarchicalLocationsView.tsx

```
@@ -229,7 +225,7 @@ export class HierarchicalLocationsView extends React.PureComponent<Props, State>
                                                         </span>
                                                     </span>
                                                     <span className="badge badge-secondary badge-pill hierarchical-locations-view__item-badge">
-                                                        {locationsByGroup[i].get(groupName)!.length}
+                                                        {(locationsByGroup[i].get(groupName) || []).length}
```

The `!` makes me think that `groupName` being a member of `locationsByGroup[i]` is an invariant that should not be violated. If that's true, then defaulting to `[]` here probably masks some other bug.

---

Yeah, I think it masks some other bug. Can you post the JS errors and the list of URIs that trigger the errors?